package.json

```diff
@@ -16,7 +16,7 @@
   },
   "husky": {
     "hooks": {
-      "pre-commit": "pretty-quick --staged ; eslint --fix ."
+      "pre-commit": "npm run format ; npm run lint:fix"
```

Comment:
`pretty-quick` didn't catch a Markdown error in the README.  I changed the pre-commit hook to run the same command as is run in CI/CD so the same checks are made to avoid linting errors getting through to the pipeline in the future.

Reply:
@zackads Surprised this was the case?  The difference between the `pretty-quick` and what is run in `CI/CD` is that `pretty-quick` will only lint your staged files which should be the preferred option as these are the only files that have been changed. This could potentially be a slight issue if you're running prettier and linting over the whole code on each commit?

I have seen [lint-staged](https://www.npmjs.com/package/lint-staged) used before if you wanted to lint both using husky.  I haven't used it before but installation instructions are:
![image](https://user-images.githubusercontent.com/76042279/105871504-4851a780-5ff1-11eb-8b1d-2846dd1ffa80.png)
But this should do prettier and linting hooks for just the staged files, which in theory should only be the ones that have changed

Reply:
Prettier being unhappy in pipeline: https://github.com/madetech/mca-beacons-webapp/runs/1769795567?check_suite_focus=true

Thinking maybe we could add the `--pattern` [flag](https://github.com/azz/pretty-quick#cli-flags) to `pretty-quick` and just give it the same list we give `prettier` 🤔

That way we can also have the `--staged` flag

Reply:
Hmm.  Unsure of how the formatting error got through in this case 🤔. 

I guess running the same formatting/linting script locally as gets run in CI/CD gives me greater confidence when I push changes that I'm not going to break the pipeline (and waste your time on a PR that's failing formatting/linting checks).  

And the performance advantage of `pretty-quick` over `npm run format` is milliseconds at this stage.  If `npm run format` starts taking a long time later in the project and the trade-off isn't worth it we can switch back to `pretty-quick`?

Reply:
Yeah, I am just a little confused in how that's managed to get through in that case...  In theory, if you're linting/formatting your staged files that should only be what you need to do to ensure the pipeline passes.

Yeah, appreciate it's better knowing what's being run locally is what is on CI, the only thing I would say, is that the linting will start to take longer once you have more files, more so than prettier.  Go with whatever you guys think is best at this stage and we can modify the scripts if we need to further down the line if need be

Reply:
@Ninamma You could do the flag.  That image I posted above for installing `lint-staged` adds the following to the `package.json` so essentially is doing that!
![image](https://user-images.githubusercontent.com/76042279/105875178-64574800-5ff5-11eb-89a0-60572c3969a4.png)

Reply:
That's a fair point!

Also not entirely sure how it got through either - might be related to node version issue maybe?
Because I tried to replicate what happened and `pretty-quick` caught the bad markdown this time.

Reply:
I've added `lint-staged` to the project in my final commit to the PR I'm about to submit, so will close this one now.  Thanks folks! 👍🏻